packages/flutter_tools/test/general.shard/dart/pub_get_test.dart

```diff
@@ -144,14 +147,67 @@ void main() {
       environment: <String, String>{'PUB_CACHE': 'custom/pub-cache/path'},
     ),
   });
+
+  testUsingContext('analytics sent on success', () async {
+    MockDirectory.findCache = true;
+    await pubGet(context: PubContext.flutterTests, checkLastModified: false);
+    verify(flutterUsage.sendEvent('pub', 'flutter-tests', label: 'success')).called(1);
+  }, overrides: <Type, Generator>{
+    ProcessManager: () => MockProcessManager(0),
+    FileSystem: () => MockFileSystem(),
+    Platform: () => FakePlatform(
+      environment: <String, String>{'PUB_CACHE': 'custom/pub-cache/path'},
```

Comment:
I would make this unmodifiable maps to reflect the Platform API, unless the FakePlatform already handles thiis

Reply:
Done